data_diff/databases/snowflake.py

```diff
@@ -35,10 +37,21 @@ def __init__(self, *, schema: str, **kw):
         logging.getLogger("snowflake.connector.network").disabled = True
 
         assert '"' not in schema, "Schema name should not contain quotes!"
-        self._conn = snowflake.connector.connect(
-            schema=f'"{schema}"',
-            **kw,
-        )
+        if (
+            "password" not in kw and "key" in kw
+        ):  # if private keys are used instead of password for Snowflake connection, read in key from path specified and pass as "private_key" to connector.
+            with open(kw.get("key"), "rb") as key:
+                p_key = serialization.load_pem_private_key(key.read(), password=None, backend=default_backend())
+
+            pkb = p_key.private_bytes(
+                encoding=serialization.Encoding.DER,
+                format=serialization.PrivateFormat.PKCS8,
+                encryption_algorithm=serialization.NoEncryption(),
+            )
+            self._conn = snowflake.connector.connect(private_key=pkb, schema=f'"{schema}"', **kw)  # replaces password
```

Comment:
if you change this to 

```python
kw['private_key'] = pkb
```

There's no need to duplicate the `connect()` function.

Reply:
Thanks, I have implemented this